src/React.Sample.Mvc4/Controllers/HomeController.cs

```diff
@@ -69,27 +71,43 @@ public HomeController()
 				new CommentModel { Author = _authors["daniel"], Text = "Foo" },
 				new CommentModel { Author = _authors["daniel"], Text = "Bar" },
 				new CommentModel { Author = _authors["daniel"], Text = "FooBarBaz" },
-		    };
-	    }
+			};
+		}
 
-        public ActionResult Index()
-        {
-            return View(new IndexViewModel
-            {
-	            Comments = _comments.Take(COMMENTS_PER_PAGE),
-				CommentsPerPage = COMMENTS_PER_PAGE
-            });
-        }
+		public ActionResult Index()
+		{
+			return View(new IndexViewModel
+			{
+				Comments = _comments.Take(COMMENTS_PER_PAGE),
+				CommentsPerPage = COMMENTS_PER_PAGE,
+				Page = 1
+			});
+		}
 
-	    public ActionResult Comments(int page)
-	    {
-		    var comments = _comments.Skip((page - 1) * COMMENTS_PER_PAGE).Take(COMMENTS_PER_PAGE);
+		[OutputCache(Duration = 0, Location = OutputCacheLocation.Any, VaryByHeader = "Content-Type")]
+		public ActionResult Comments(int page)
+		{
+			Response.Cache.SetOmitVaryStar(true);
```

Comment:
Do you man the line about omit vary by star? there is a pretty esoteric reason why iis won't send the cache header to the response when you use varybyheader. I don't understand it fully, but testing the examples given by this SO thread worked
http://stackoverflow.com/questions/7849392/unable-to-append-vary-header-to-response